client/src/components/ratingsReviews/NewReview.jsx

```diff
@@ -1,18 +1,326 @@
 import React from 'react';
+import Modal from '../shared/Modal.jsx';
 import './newReview.css';
 
-const NewReview = (props) => {
-  const { reviews } = props;
+class NewReview extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      show: false,
+      modalTitle: 'Write Your Review',
+      modalSubtitle: 'About the ',
+      selectedProduct: this.props.selectedProduct,
+      formName: 'Review',
+      overallRating: 0,
+      recommend: false,
+      characteristics: {},
+      reviewSummary: '',
+      reviewBody: '',
+      photos: [],
+      nickname: '',
+      email: '',
+      requires: {
+        overallRating: '',
+        recommend: '',
+        characteristics: '',
+        reviewBody: '',
+        nickname: '',
+        email: '',
+      }
+    };
 
-  const handleAddReview = () => {
-    console.log('Add Clicked');
+    this.key = 0;
+
+    this.radio = {
+      Size: {
+        one: 'A size too small',
+        two: '1/2a size too small',
+        three: 'Perfect',
+        four: '1/2a size too big',
+        five: 'A size too wide'
+      },
+      Width: {
+        one: 'Too narrow',
+        two: 'Slightly narrow',
+        three: 'Perfect',
+        four: 'Slightly wide',
+        five: 'Too wide'
+      },
+      Comfort: {
+        one: 'Uncomfortable',
+        two: 'Slightly uncomfortable',
+        three: 'OK',
+        four: 'Comfortable',
+        five: 'Perfect'
+      },
+      Quality: {
+        one: 'Poor',
+        two: 'Bellow average',
+        three: 'What I expected',
+        four: 'Pretty great',
+        five: 'Perfect'
+      },
+      Length: {
+        one: 'Runs short',
+        two: 'Runs slightly short',
+        three: 'Perfect',
+        four: 'Runs slightly long',
+        five: 'Runs long'
+      },
+      Fit: {
+        one: 'Runs tight',
+        two: 'Runs slightly tight',
+        three: 'Perfect',
+        four: 'Runs slightly long',
+        five: 'Runs long'
+      },
+    };
+
+    this.radioArr = Object.entries(this.radio);
+    console.log('radioArr:', this.radioArr);
+
+    this.recommendArr = ['Yes', 'No'];
+
+    this.showModal = this.showModal.bind(this);
+    this.hideModal = this.hideModal.bind(this);
+  }
+
+  showModal = () => {
+    this.setState({ show: true });
   };
 
-  return (
-    <div id='rr-new-review'>
-      <div id='rr-add-review' class='button uppercase' onClick={handleAddReview}>ADD A REVIEW<div class='plus'>+</div></div>
-    </div>
-  );
-};
+  hideModal = () => {
+    this.setState({ show: false });
+  }
+
+  handleOnChange(e) {
+    let id = e.target.id;
+    if (id === 'reviewSummary' || id === 'reviewBody' || id === 'email' || id === 'nickname' || id === 'overallRating' || id === 'recommend') {
```

Comment:
I'm not sure, If we put all id in array and then loop through array, is that better than using `||`